pennylane_orquestra/orquestra_device.py

```diff
@@ -0,0 +1,378 @@
+        """
+        qasm_str = circuit.to_openqasm(rotations=not self.analytic)
+
+        qasm_without_measurements = re.sub("measure.*?;\n", "", qasm_str)
+        return qasm_without_measurements
+
+    def process_observables(self, observables):
+        """Processes the observables provided with the circuits.
+
+        If the observable defined is the identity, then no serialization
+        happens. Instead, the index of the observable is saved.
+
+        Args:
+            observables (list): a list of observables to process
+
+        Returns:
+            tuple:
+
+                * the serialized non-identity operators
+                * the indices of the identity operators
+        """
+        ops = []
+        identity_indices = []
+        for idx, obs in enumerate(observables):
+            if not isinstance(obs, Identity):
+                # Only serialize if it's not the identity
+                ops.append(self.serialize_operator(obs))
+            else:
+                # Otherwise keep track of the indices and use the theoreticaly
+                # value as a result later
+                identity_indices.append(idx)
+
+        return ops, identity_indices
+
+    def serialize_operator(self, observable):
+        """
+        Serialize the observable specified for the circuit as an OpenFermion
+        operator.
+
+        Args:
+            observable (~.Observable): the observable to get the operator
+                representation for
+
+        Returns:
+            str: string representation of terms making up the observable
+        """
+        if not self.analytic:
+            obs_wires = observable.wires
+            wires = self.wires.indices(obs_wires)
+            op_str = self.pauliz_operator_string(wires)
+        else:
+            op_str = self.qubit_operator_string(observable)
+
+        return op_str
+
+    @staticmethod
+    def pauliz_operator_string(wires):
+        """Creates an OpenFermion operator string based on the related wires
+        that can be passed when creating an ``openfermion.IsingOperator``.
+
+        This method is used if rotations are needed for the backend specified.
+        In such a case a string that represents measuring PauliZ on each of the
+        affected wires is used.
+
+        **Example**
+
+        >>> dev = QeQiskitDevice(wires=2)
+        >>> wires = [0, 1, 2]
+        >>> op_str = dev.pauliz_operator_string(wires)
+        >>> print(op_str)
+        [Z0 Z1 Z2]
+        >>> print(openfermion.IsingOperator(op_str))
+        1.0 [Z0 Z1 Z2]
+
+        Args:
+            wires (Wires): the wires the observable of the quantum function
+                acts on
+
+        Returns:
+            str: the ``openfermion.IsingOperator`` string representation
+        """
+        op_wires_but_last = [f"Z{w} " for w in wires[:-1]]
+        op_last_wire = f"Z{wires[-1]}"
+        op_str = "".join(["[", *op_wires_but_last, op_last_wire, "]"])
+        return op_str
```

Comment:
Why do the last wire separately? Wouldn't just `""join([f"Z{w} " for w in wires])` work equally well? :thought_balloon:

Reply:
No space is added after the last wire. Adding a comment for clarification.

Reply:
Oh, I see. :+1: